OgreMain/include/OgreSceneLoader.h

```diff
@@ -0,0 +1,18 @@
+#ifndef __SCENELOADER_H__
+#define __SCENELOADER_H__
+
+#include "OgreSceneManager.h"
+
+namespace Ogre {
+    
+    class _OgreExport SceneLoader
+    {
+    public:
+        virtual ~SceneLoader() {};
+        virtual bool load(const String& filename, const String& groupname, SceneNode *rn);
+        virtual bool load(const String& filename, const String& groupname, SceneManager *sm);
```

Comment:
SceneNodes know their creator, so this one is not needed

Reply:
Also I would prefer exceptions for error handling here

Reply:
This function should be pure virtual as not overriding it makes no sense. (making this class header only)

Reply:
should take DataStream instead of a filename for flexibility. See `ScriptLoader::parseScript`. Also adapt the comments from there.